Add user CRUD

EvgeniiMaiorov opened the pull request:
https://trello.com/c/BWzKXpBK/6-add-gem-jsonapi-serializer

Added gem 'jsonapi-serializer' for using in API endpoints.
Added user CRUD.

Commit:
Add gem 'jsonapi-serializer'

anevsevra commented on app/controllers/api/v1/users_controller.rb:
```diff
@@ -0,0 +1,57 @@
+
+module Api
+  module V1
+    class UsersController < Api::ApplicationController
```
It is better to create base controller in the scope of each API version. Because there can be significant changes between API versions that apply to base controllers. For example, the way how request is authenticated.

anevsevra replied:
Also it is good to restrict your controllers regarding requested format. In your case allow only json.

anevsevra commented on config/routes.rb:
```diff
@@ -2,4 +2,12 @@
+
+  namespace :api do
+    namespace :v1 do
+      resources :users
```
Exclude `:new` and `:edit` actions as they are not used.

Commit:
Change routes

Add format constraint to :api namespace
Move application_controller from Api module to Api::V1 module
Add exclude for new and edit actions in resource :users
